functions/__fzf_preview_file.fish

```diff
@@ -1,17 +1,26 @@
 # helper function for __fzf_search_current_dir
 function __fzf_preview_file --description "Prints a preview for the given file based on its file type." --argument-names file_path
-    set file_type (file --brief -i "$file_path")
+    if test -d "$file_path"
+        set file_type "d"
+    else
+        set file_type (ls -o "$file_path" | cut -c 1) #exa is faster than ls and file
+    end
     switch $file_type
-        case "directory"
+        case "d"
             # Setting CLICOLOR_FORCE forces colors to be enabled even to a non-terminal output
             CLICOLOR_FORCE=true ls -a "$file_path"
-        case "socket"
+        case "s"
```

Comment:
Good to know exa tells you if something is a socket. Thanks for trying your best to preserve the intended behavior!

Reply:
+ Both ls and exa would suffice for this.
+ [Here](https://en.wikipedia.org/wiki/Unix_file_types) have a look, if you want special use case for each type of file, its possible too.
+ You do not need exa, but if you want I can push exa file for you. exa, in general, is prettier and faster than ls.